gtfs-realtime/spec/en/reference.md

```diff
@@ -133,7 +133,9 @@ Depending on the value of ScheduleRelationship, a TripUpdate can specify:
 
 The updates can be for future, predicted arrival/departure events, or for past events that already occurred. In most cases information about past events is a measured value thus its uncertainty value is recommended to be 0\. Although there could be cases when this does not hold so it is allowed to have uncertainty value different from 0 for past events. If an update's uncertainty is not 0, either the update is an approximate prediction for a trip that has not completed or the measurement is not precise or the update was a prediction for the past that has not been verified after the event occurred.
 
-When multiple trips in one block are included in one feed, the respective TripUpdate entities are not required to be added to the feed in the same order that they are scheduled in the block (for more information about trips and blocks, please refer to [GTFS trips.txt](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#tripstxt)). For example, if there are trips with `trip_ids` 1, 2, and 3 that all belong to one block, and the vehicle travels trip 1, then trip 2, and then trip 3, the `trip_update` entities may appear in any order - for example, adding trip 2, then trip 1, and then trip 3 is allowed.
+If a vehicle is serving multiple trips within the same block (for more information about trips and blocks, please refer to [GTFS trips.txt](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#tripstxt)):
+* the feed should include a TripUpdate for the trip currently being served by the vehicle as well as a another TripUpdate for the next trip in the block. Including multiple TripUpdates for the same vehicle avoids prediction "pop-in" for riders as the vehicle transitions from one trip to another and also gives riders advance notice of delays that impact downstream trips (e.g., when the known delay exceeds planned layover times between trips).
```

Comment:
This sentence is limiting the propagation to the next trip.
We could open the door to including more than one continuation trip. 
If the trips are very short, that will be useful.

Reply:
@juanborre Yes, I think this is a key part of this proposal to decide on. We need to balance suggesting providing more than one TripUpdate, without encouraging producers to share too many TripUpdates for future trips to the point where predictions are meaningless.

The current text was an attempt to explicitly suggest providing predictions for the next trip (Trip B, which usually makes sense), without explicitly addressing additional trips - so my intent wasn't to prohibit Trips C, D etc. if they make sense in a certain context.

Do you have any suggestions for different language that strikes this balance?

Reply:
If past trips in block continue to be included as TripUpdate entities, is this allowed?  Purpose for this is for block-based systems to deterministically verify conclusion of full service.

Reply:
@mpaine-act Yes, my new text is not intended to prohibit the inclusion of past TripUpdates in any way. In fact, there is text just above and below the text I added that says:

>The updates can be for future, predicted arrival/departure events, or for past events that already occurred. 
>...
>Note that the update can describe a trip that has already completed... Note that the feed provider is allowed, but not required, to purge past updates

Reply:
I understand. On the same line of thinking, providing a prediction for the immediate next trip can be as meaningless as providing a prediction for the trip N+3. It will depend on the duration of the trips and how far in the future we are.

How about including reliability in the phrasing, like:
> The feed should include a TripUpdate for the trip currently being served by the vehicle. If the producer is able to provide reliable real time information for the next trips that the vehicle is doing in the block, they are encouraged to include TripUpdates for those trips. The producer should only provide updates which they are sure of.

Reply:
We like the changes in the revision from @juanborre , notably that multiple TripUpdates "are encouraged" rather than "should" be included and the emphasis on providing more than one only if producer is confident of the predictions for future trips. Here is a suggested revision:
 

> The feed should include a TripUpdate for the trip currently being served by the vehicle. Producers are encouraged to include TripUpdates for one or more trips after the current trip in this vehicle's block if the producer is confident in the quality of the predictions for these future trip(s).

Reply:
@juanborre and @ibi-group-team Thanks for the suggestions! I've revised the text as suggested in @ibi-group-team's [latest post](https://github.com/google/transit/pull/206#discussion_r385863796) in https://github.com/google/transit/pull/206/commits/bb8a727cbb060af23bb72e94e62c97a32f8f5b68.